lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -830,6 +830,17 @@ defmodule Kernel.ErrorsTest do
       'if true do\n  foo = [],\n  baz\nend'
   end
 
+  # As reported and discussed in
+  # https://github.com/elixir-lang/elixir/issues/4419.
+  test "characters literal are printed correctly in syntax errors" do
+    assert_compile_fail SyntaxError,
+      "nofile:1: syntax error before: ?a",
+      ':ok ?a'
+    assert_compile_fail SyntaxError,
+      "nofile:1: syntax error before: ?\\s",
```

Comment:
Let's also add a test for one unicode character with high value to ensure the "escaping" will also work as expected there and then we can :shipit:. :)